care/facility/tests/test_facility_notification_api.py

```diff
@@ -0,0 +1,128 @@
+from enum import Enum
+
+from rest_framework import status
+from rest_framework.test import APIRequestFactory, APITestCase
+from rest_framework_simplejwt.tokens import RefreshToken
+
+from care.utils.tests.test_utils import TestUtils
+
+
+class ExpectedListNotitificationKeys(Enum):
+    id = "id"
+    event = "event"
+    caused_by = "caused_by"
+    message = "message"
+    created_date = "created_date"
+    read_at = "read_at"
+
+
+class ExpectedListCausedByKeys(Enum):
+    id = "id"
+    first_name = "first_name"
+    last_name = "last_name"
+    user_type = "user_type"
+    username = "username"
+    email = "email"
+    last_login = "last_login"
+
+
+class ExpectedRetrieveNotificationKeys(Enum):
+    id = "id"
+    event = "event"
+    caused_by = "caused_by"
+    event_type = "event_type"
+    created_date = "created_date"
+    read_at = "read_at"
+    message = "message"
+    caused_objects = "caused_objects"
+
+
+class ExpectedRetrieveCausedByKeys(Enum):
+    id = "id"
+    first_name = "first_name"
+    username = "username"
+    email = "email"
+    last_name = "last_name"
+    user_type = "user_type"
+    last_login = "last_login"
+
+
+class NotificationViewSetTestCase(APITestCase, TestUtils):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.factory = APIRequestFactory()
+        cls.notification = cls.create_notification()
+
+    def setUp(self) -> None:
+        # Refresh token to header
+        refresh_token = RefreshToken.for_user(self.user)
+        self.client.credentials(
+            HTTP_AUTHORIZATION=f"Bearer {refresh_token.access_token}"
+        )
```

Comment:
we are already doing force authenticate with self.user so this won’t be required